core/io/resource_loader.cpp

```diff
@@ -49,6 +49,11 @@ Ref<ResourceFormatLoader> ResourceLoader::loader[ResourceLoader::MAX_LOADERS];
 int ResourceLoader::loader_count = 0;
 
 bool ResourceFormatLoader::recognize_path(const String &p_path, const String &p_for_type) const {
+	bool ret;
```

Comment:
Should be initialized as we don't have guarantees that the gdvirtual call will do it:
```
D:\a\godot\godot\core\io\resource_loader.cpp(54) : warning C4701: potentially uninitialized local variable 'ret' used
```

Reply:
Ah ok, I'll fix that but it seems there are a lot of other places where this is done that way. I just copy-pasted from elsewhere.

Reply:
You likely started working off an outdated branch, this was fixed in #66808.

Reply:
Not really, see https://github.com/godotengine/godot/blob/master/servers/text/text_server_extension.cpp#L670 or https://github.com/godotengine/godot/blob/master/scene/resources/style_box.cpp#L38 for example.
I guess not all of those trigger a warning ? Not sure why though.

Reply:
Weird, maybe `float`/`double` are not default initialized by Visual Studio so it doesn't complain about those.

Reply:
Well, I don't know, but some bools are still uninitialized: https://github.com/godotengine/godot/blob/master/scene/resources/style_box.cpp#L46 :shrug: 

That's weird, but well, the compiler has its reasons that reason ignores.